File: day5/part2.py
# -*- coding: utf-8 -*-

################ [INFINITE LOOP] ################

if __name__ == "__main__":
    with open("input") as f:
        lines = f.readlines()
    
    res = 0
    seeds = lines[0].split(":")[1].split(" ")
    seeds.pop(0)
    seeds = [int(x) for x in seeds]
    lines.pop(0)

    start = 0
    for init_seed in seeds:
        if not start: start = init_seed
        else:
            for i in range(init_seed):
                seed = start + i
                print(f"compute {seed}")
                converted = False
                for line in lines:
                    if "map" in line or line == "\n":
                        converted = False
                    elif not converted:
                        if int(line.split(" ")[1]) <= seed <= int(line.split(" ")[1]) + int(line.split(" ")[2]):
                            seed = (seed - int(line.split(" ")[1])) + int(line.split(" ")[0])
                            converted = True
                
                if seed < res or res == 0:
                    res = seed
                print(res)
            start = 0
    
    print(f"Result of the puzzle: {res}")

File: day9/part2.js
const fs = require('fs');

const input = fs.readFileSync('./input', 'utf8').split('\n');
input.pop();

var res = 0;

input.forEach(line => {
    var next = [[]];
    next[0] = line.split(' ').map(Number);
    var i = 0;
    var finished = false;
    while (!finished) {
        next[i+1] = [];
        for (var j = 0; j < next[i].length - 1; j++) {
            next[i+1][j] = next[i][j+1] - next[i][j];
        }
        finished = true;
        next[i+1].forEach(element => { if (element != 0) finished = false; });
        i++;

    }

    next[i] = [0, ...next[i]];
    finished = false;
    while (i !== 0) {
        next[i-1] = [next[i-1].slice(0)[0] - next[i].slice(0)[0], ...next[i-1]];
        i--;        
    }

    res += next[0].slice(0)[0];
});


console.log("Result of the puzzle: " + res);